FreeRTOS/FreeRTOS-Plus-TCP
# [V4.3.0] Update changelog and version number

Description
-----------
This PR updates the changelog and version number in preparation for the V4.3.0 release.
Add script to update the version number in source files as part of the auto-release process, similar to other repos [example, [here](https://github.com/FreeRTOS/coreMQTT/blob/main/.github/workflows/release.yml#L79-L90)]

Test Steps
-----------
Tested with the changes in fork: https://github.com/tony-josi-aws/FreeRTOS-Plus-TCP/actions/runs/12252149023

Checklist:
----------


- ~[] I have modified and/or added unit-tests to cover the code changes in this Pull Request.~

Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
